wandb/sdk/artifacts/artifact.py

```diff
@@ -501,6 +482,57 @@ def metadata(self, metadata: dict) -> None:
         """
         self._metadata = self._normalize_metadata(metadata)
 
+    @property
+    def ttl(self) -> Union[datetime.timedelta, None]:
+        """Artifact TTL(time to live).
+
+        Set a TTL to mark the artifact for deletion after the given duration from Artifact creation has passed.
+        To turn off previously set artifact TTL, reset TTL to wandb.ArtifactTTL.DEFAULT.
+
+        Examples:
+            Basic usage
+            ```
+            artifact.ttl = timedelta(days=10) # artifact will be marked for deletion on artifact.created_at + 10 days
+            artifact.save() # or log the artifact if its new
+            ```
+        """
+        if self._ttl_changed and self._ttl_is_inherited:
+            raise Exception(
+                "Unable to fetch Inherited TTL until artifact is logged/saved"
+            )
+        if self._ttl_duration_seconds is None:
+            return None
+        return datetime.timedelta(seconds=self._ttl_duration_seconds)
+
+    @ttl.setter
+    def ttl(self, ttl: Union[datetime.timedelta, ArtifactTTL, None]) -> None:
+        """Artifact TTL(time to live).
+
+        Note:
+            This ttl is set on the artifact version level.
+
+        Arguments:
+            ttl: How long the artifact will remain active from its creation. timedelta must be positive,
+            To reset artifact TTL to inherit from collection level rules `artifact.ttl = wandb.ArtifactTTL.INHERIT`
+        """
+        if self.type == "wandb-history":
+            raise ValueError("Cannot set artifact TTL for type wandb-history")
+
+        self._ttl_changed = True
+        if isinstance(ttl, ArtifactTTL):
+            if ttl == ArtifactTTL.INHERIT:
+                self._ttl_is_inherited = True
```

Comment:
Nit: no need for the outer if. In fact, when someone adds a new item to them enum, we don't want this function to silently do nothing. In fact, many linters enforce this, see [here](https://www.google.com/search?q=switch+statement+exhaustiveness).
```suggestion
        if ttl == ArtifactTTL.INHERIT:
            self._ttl_is_inherited = True
```

Reply:
It was giving my mypy errors when I didn't have the isinstance check so i switched it back. it freaks out when you do ttl timedelta stuff in the lower if statement without ruling out its an enum. I can just add a statement to else: error for invalid enum

Reply:
I see. Then how about we add:
```
else:
    raise NotImplementedError(ttl)
```